packages/core/src/domain/configuration.ts

```diff
@@ -215,18 +227,42 @@ function getEndpoint(type: string, conf: TransportConfiguration, source?: string
   return `https://${host}/v1/input/${conf.clientToken}?${parameters}`
 }
 
-export function isIntakeRequest(url: string, configuration: Configuration) {
-  return (
-    getPathName(url).indexOf('/v1/input/') !== -1 &&
-    (haveSameOrigin(url, configuration.logsEndpoint) ||
-      haveSameOrigin(url, configuration.rumEndpoint) ||
-      haveSameOrigin(url, configuration.traceEndpoint) ||
-      (!!configuration.internalMonitoringEndpoint && haveSameOrigin(url, configuration.internalMonitoringEndpoint)) ||
-      (!!configuration.replica &&
-        (haveSameOrigin(url, configuration.replica.logsEndpoint) ||
-          haveSameOrigin(url, configuration.replica.rumEndpoint) ||
-          haveSameOrigin(url, configuration.replica.internalMonitoringEndpoint))))
-  )
+function getHost(type: EndpointType, conf: TransportConfiguration) {
+  if (conf.useNewIntakeDomains && NEW_INTAKE_DOMAIN_ALLOWED_SITES.indexOf(conf.site) !== -1) {
+    return `${type}.browser-intake-${conf.site}`
+  }
+  const oldTypes = {
+    [EndpointType.LOGS]: 'browser',
+    [EndpointType.RUM]: 'rum',
+    [EndpointType.TRACE]: 'public-trace',
+  }
+  return `${oldTypes[type]}-http-intake.logs.${conf.site}`
+}
+
+function getIntakeUrls(conf: TransportConfiguration, withReplica: boolean) {
+  if (conf.proxyHost) {
+    return [`https://${conf.proxyHost}/v1/input/`]
+  }
+  const sites = [conf.site]
+  if (conf.buildMode === BuildMode.STAGING && withReplica) {
+    sites.push(INTAKE_SITE[Datacenter.US])
+  }
+  const urls = []
+  for (const site of sites) {
+    urls.push(
+      `https://rum-http-intake.logs.${site}/v1/input/`,
+      `https://browser-http-intake.logs.${site}/v1/input/`,
+      `https://public-trace-http-intake.logs.${site}/v1/input/`
+    )
+    if (conf.useNewIntakeDomains && NEW_INTAKE_DOMAIN_ALLOWED_SITES.indexOf(site) !== -1) {
```

Comment:
Since this will be a customer configuration, it seems unlikely that they will perform it on one SDK but not on another.
IMO, we could then check only one domains set or the other depending on the conf.

Reply:
Makes sense. But what if they used a site that doesn't support new intake domains yet (like EU right now), don't we want to fallback to the old ones? Should we show them a warning in that case?

Reply:
> what if they used a site that doesn't support new intake domains yet (like EU right now), don't we want to fallback to the old ones?

Yes

> Should we show them a warning in that case?

why not, we do this kind of checks on the configuration already


I don't see how these questions are related to the initial comment though.

Reply:
Sorry, I misunderstood your initial comment. I thought that you suggested to only check the configuration option and choose domains based on it. But now I see that we also check if the configured site is supported. All good now.